### DISCUSSION TITLE
Feature request: A field for "geographic location" and a "map" view for geo-located projects.

### DISCUSSION THREAD

**Discussion by hopezh** [marcusolsson/obsidian-projects#690] [General]
For managing projects based on geographic location, a "geolocation" field and a "map" view will be very helpful to visualize the spatial distribution of all projects.

The "geolocation" field could be composed of two decimal numbers indicating the latitude and longitude of the project, or a url to its position on the google map.

The "map" view could use either Google map or MapBox as the basic map.

Hope the team can consider this feature.

**Comment by Acylation** [comment]
It could be a useful feature, as I have experienced from the map view of Markwhen. However, due to constraints in resources and time, prioritizing the development of a new view might not be feasible at the moment. I'll mark this as an idea for the future consideration anyway.

It’s worth mentioning that the views in the plugin are separate components and are not coupled with the data source. The plugin also exposed ts API for buiding custom views (https://github.com/marcusolsson/obsidian-projects/blob/main/obsidian-projects-types/README.md). If you’re interested in this, and have familiarity with geographic APIs, I’d be more than happy to provide assistance.

Quote reply for new views here: https://github.com/marcusolsson/obsidian-projects/issues/6#issuecomment-1481904610
> Views are isolated parts of the plugin. Each view acts almost like a plugin within the Projects plugin. I believe this makes it easier for other contributors, as you can build new views and improve existing ones without having to understand the rest of the Projects codebase.
>

**Reply by Acylation** [reply]
A geo plugin for Obsidian, hopefully it would have the capability to integrate with Projects. https://github.com/esm7/obsidian-map-view
